protocol.md

```diff
@@ -0,0 +1,338 @@
+Status Secure And Decentralized Messaging Protocol
+==================================================
+
+- [Status Secure And Decentralized Messaging Protocol](#status-secure-and-decentralized-messaging-protocol)
+- [Abstract](#abstract)
+- [Introduction](#introduction)
+- [Terminology](#terminology)
+- [Basic Assumption](#basic-assumption)
+- [Protocol Overview](#protocol-overview)
+- [Payload](#payload)
+	- [Content types](#content-types)
+	- [Message types](#message-types)
+	- [Clock vs Timestamp And Message Ordering](#clock-vs-timestamp-and-message-ordering)
+	- [Quoting](#quoting)
+- [Whisper adapter](#whisper-adapter)
+	- [Whisper node configuration](#whisper-node-configuration)
+	- [Keys management](#keys-management)
+	- [Encryption algorithms](#encryption-algorithms)
+	- [Topic](#topic)
+	- [Message encryption](#message-encryption)
+- [Perfect Forward Secrecy (PFS)](#perfect-forward-secrecy-pfs)
+- [Device syncing](#device-syncing)
+- [One-to-one messages](#one-to-one-messages)
+	- [Sending](#sending)
+		- [Sending using PFS](#sending-using-pfs)
+	- [Receiving](#receiving)
+- [Public messages](#public-messages)
+	- [Sending](#sending-1)
+	- [Receiving](#receiving-1)
+- [Group messages](#group-messages)
+- [Offline messages](#offline-messages)
+	- [Anonymity concerns](#anonymity-concerns)
+- [Whisper V6 extensions (or Status Whisper Node)](#whisper-v6-extensions-or-status-whisper-node)
+	- [New RPC methods](#new-rpc-methods)
+
+# Abstract
+
+Ethereum empowers users and developers to interact with totally new kind of applications called Dapps (Decentralized applications). These application allows to interact with the blockchain on a completely new level which is not only about exchanging values but also executing arbitrary logic. This logic can form very sophisticated programs like DAOs (Decentralized autonomous organizations). The missing part here is how users of Dapps can communicate securely and in a decentralized way with each other. Communication is an essential part of any activity. In this document, we specify a secure and decentralized messaging protocol that is capable of running on the Ethereum network.
+
+# Introduction
+
+TBD
+
+# Terminology
```

Comment:
Might want to define a section on crypto primitives etc, there's a section on this in PFS afaik

Reply:
Question if it's needed. The protocol itself relies on Whisper spec entirely and it's kind of out of our control. We should link to Whisper spec.

Reply:
Whisper spec is not good enough imo, and even so we want to at least abbreviate. I.e. Double Ratchet spec good enough but still take gist of it (primitives and what guarantees it provides, and some on flow)